src/renderers/dom/fiber/ReactDOMFiberEntry.js

```diff
@@ -368,7 +368,13 @@ var DOMRenderer = ReactFiberReconciler({
       const hostContextDev = ((hostContext: any): HostContextDev);
       validateDOMNesting(null, text, null, hostContextDev.ancestorInfo);
     }
-    var textNode: TextInstance = document.createTextNode(text);
+
+    var doc = rootContainerInstance;
+    if (rootContainerInstance.nodeType !== DOCUMENT_NODE && doc.ownerDocument) {
```

Comment:
PS. `rootContainerInstance` should changed to `doc` for consistency.

Reply:
That's a good call out, but I'm actually removing that ownerDocument and changing the name for short.